### PR TITLE
Use Python 3.11 for upstream and doctests CI build

### PR DESCRIPTION
**Pull request by graingert**
This doesn't fix https://github.com/dask/dask/issues/10593 as pandas is not compatible with numpy v2 again due to changes from NEP50

https://github.com/dask/dask/pull/10594 extended with fixes for https://github.com/pandas-dev/pandas/pull/55636 and fixes for the doctests

### CODE REVIEW

**Comment by graingert** [comment]
this looks like it's got a lot of failures now as a result of NEP 50

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -849,9 +849,9 @@ def get_partition(self, n):
         Dask DataFrame Structure:
                          name     id        x        y
         npartitions=1
-        2021-01-01     object  int64  float64  float64
+        2021-01-01     string  int64  float64  float64
         2021-01-02        ...    ...      ...      ...
-        Dask Name: get-partition, 2 graph layers
+        Dask Name: get-partition, 3 graph layers
```
Comment (graingert) [inline dask/dataframe/core.py:854]: why does this gain an extra layer?

Comment (jrbourbeau) [reply dask/dataframe/core.py:854]: There's an extra layer for the automatic pyarrow string conversion

**Review by jrbourbeau — APPROVED** [review]
Thanks @graingert! This is in

**Pull request merged**